File: bundler/solc/src/utils/invariant.js
/**
 * Asserts an invariant in a typesafe way
 * @param {boolean} condition - The condition to assert
 * @param {string} message - The message to display if the condition is false
 * @throws {Error} Throws an error if the condition is false
 */
export function invariant(condition, message) {
	if (!condition) {
		throw new Error(message)
	}
}

File: resolutions-go/utils/invariant.go
package utils

import (
	"github.com/williamcory/tevm/go-claude/resolutions-go"
)

// Invariant asserts an invariant in a typesafe way
func Invariant(condition bool, message string) {
	if !condition {
		panic(resolutions.InvariantError{Message: message})
	}
}